filecoin-proofs/src/bin/phase2.rs

```diff
@@ -141,22 +257,54 @@ fn main() {
         .subcommand(new_command)
         .subcommand(contribute_command)
         .subcommand(verify_command);
+
     let matches = app.get_matches();
 
     match matches.subcommand() {
         (command, Some(mm)) => {
             let params_path = value_t!(mm, "parameters", String)
                 .expect("Could not convert `parameters` CLI argument to string");
 
+            let sector_size: u64 = mm
+                .value_of("sector-size")
+                .map(|s| match s.to_lowercase().as_ref() {
+                    "1kib" => SECTOR_SIZE_ONE_KIB,
+                    "16mib" => SECTOR_SIZE_16_MIB,
+                    "256mib" => SECTOR_SIZE_256_MIB,
+                    "1gib" => SECTOR_SIZE_1_GIB,
+                    "32gib" => SECTOR_SIZE_32_GIB,
+                    _ => unreachable!("Invalid CLI `sector-size` argument"),
+                })
+                .unwrap();
+
+            let porep = mm.is_present("porep");
+            let post = mm.is_present("post");
+
+            if !(porep ^ post) {
+                panic!("Please specify one and only one of the CLI flags `--porep` or `--post`");
```

Comment:
you can use https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.conflicts_with instead